substrafl/remote/register/manage_dependencies.py

```diff
@@ -0,0 +1,206 @@
+"""
+Utility functions to manage dependencies (building wheels, compiling requirement...)
+"""
+import logging
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+from pathlib import Path
+from typing import List
+
+from substrafl.dependency import Dependency
+from substrafl.exceptions import IncompatibleDependenciesError
+from substrafl.exceptions import InvalidUserModuleError
+
+logger = logging.getLogger(__name__)
+
+LOCAL_WHEELS_FOLDER = Path.home() / ".substrafl"
+
+
+def build_user_dependency_wheel(lib_path: Path, operation_dir: Path) -> str:
+    """Builds the wheel for user dependencies passed as a local module.
+
+    Args:
+        lib_path (Path): where the module is located
+        operation_dir (Path): where the wheel needs to be copied
+
+    Returns:
+        str: the filename of the wheel
+    """
+    # sys.executable takes the Python interpreter run by the code and not the default one on the computer
+    try:
+        ret = subprocess.run(
+            [
+                sys.executable,
+                "-m",
+                "pip",
+                "wheel",
+                str(lib_path) + "/",
+                "--no-deps",
+            ],
+            cwd=str(operation_dir),
+            check=True,
+            capture_output=True,
+            text=True,
+        )
+    except subprocess.CalledProcessError as e:
+        raise InvalidUserModuleError(e.stdout + e.stderr)
```

Comment:
Shouldn't we create two attributes (e.g. `stdout` and `stderr`) so the user could get the output independently?

Reply:
Should `InvalidUserModuleError` inherit from `subprocess.CalledProcessError` so that these fields are available? 🤔